cocos/2d/components/rich-text.ts

```diff
@@ -684,19 +688,21 @@ export class RichText extends Component {
 
             curString = longStr.substring(curStart, curEnd);
             leftString = longStr.substring(curEnd);
-            leftStringSize = this._calculateSize(styleIndex, leftString);
+            leftStringSize.set(this._calculateSize(styleIndex, leftString));
+            curStringSize.set(this._calculateSize(styleIndex, curString));
 
             leftTryTimes--;
 
             // Exit: If the left part string size is less than 2048, the method will finish.
-            if (leftStringSize.x < 2048) {
+            if (leftStringSize.x < 2048 && curStringSize.x < sizeForOnePart) {
+                partStringArr.push(curString);
                 curStart = text.length;
                 curEnd = text.length;
                 curString = leftString;
-                partStringArr.push(curString);
+                if (leftString !== '') {
+                    partStringArr.push(curString);
```

Comment:
why push curString twice?

Reply:
The logic here is to jump out of the loop when the remaining width is less than 2048, at this time, the penultimate line and the last line should be pushed in, the previous bug is missing the penultimate line